### PR TITLE
Add metrics to performance tests

### PR DESCRIPTION
**Pull request by pb8o**
## Changes

Convert performance tests using TestMatrix to new fixtures, so we can assign metric dimensions. They also result in simpler, shorter tests.

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

~- [ ] If a specific issue led to this PR, this PR closes the issue.~
- [x] The description of changes is clear and encompassing.
~- [ ] Any required documentation changes (code and docs) are included in this PR.~
~- [ ] API changes follow the [Runbook for Firecracker API changes][2].~
~- [ ] User-facing changes are mentioned in `CHANGELOG.md`.~
- [x] All added/changed functionality is tested.
~- [ ] New `TODO`s link to an issue.~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

~- [ ] This functionality can be added in [`rust-vmm`][1].~

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: .buildkite/pipeline_perf.py [hunk 1]
```diff
@@ -10,21 +10,12 @@
...
-            "devtool_opts": "-r 16834m -c 1-10 -m 0",
-            "kernels": ["linux_5.10"],
-            "timeout_in_minutes": 120,
-        },
-    ],
+    "block": {
+        "label": "🖴 Block Performance",
+        "test_path": "integration_tests/performance/test_block_performance.py",
+        "devtool_opts": "-r 16834m -c 1-10 -m 0",
+        "timeout_in_minutes": 240,
```
Comment (kalyazin) [inline .buildkite/pipeline_perf.py:17]: Is this because it is now a cumulative timeout for 2 tests?

Comment (pb8o) [reply .buildkite/pipeline_perf.py:17]: Yep

File: tests/integration_tests/performance/test_vsock_throughput.py [hunk 2]
```diff
@@ -268,100 +263,59 @@ def pipes(basevm, current_avail_cpu, env_id):
...
-        monitor_memory=False,
-    )
-    basevm = vm_instance.vm
-    # Create a vsock device
-    basevm.vsock.put(vsock_id="vsock0", guest_cid=3, uds_path="/" + VSOCK_UDS_PATH)
 
-    basevm.start()
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn()
+    vm.basic_config(vcpu_count=vcpus, mem_size_mib=1024)
```
[the remark this answers is not in the document]
Comment (pb8o) [reply tests/integration_tests/performance/test_vsock_throughput.py:277]: Will do.

File: tests/integration_tests/performance/test_network_latency.py [hunk 3]
```diff
@@ -130,115 +125,68 @@ def consume_ping_output(cons, raw_data, requests):
...
-        ssh_key=ssh_key,
-        config=context.microvm,
-        monitor_memory=False,
-    )
-    basevm = vm_instance.vm
-    basevm.start()
+    # Create a microvm from artifacts
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn()
+    vm.basic_config(vcpu_count=1, mem_size_mib=1024)
```
Comment (kalyazin) [inline tests/integration_tests/performance/test_network_latency.py:144]: Nit: There is one, sorry.

File: tests/integration_tests/performance/test_network_tcp_throughput.py [hunk 4]
```diff
@@ -308,100 +304,62 @@ def pipes(basevm, host_ip, current_avail_cpu, env_id):
...
-    )
-    basevm = vm_instance.vm
-    basevm.start()
-    custom = {
-        "microvm": context.microvm.name(),
-        "kernel": context.kernel.name(),
-        "disk": context.disk.name(),
+    vm = microvm_factory.build(guest_kernel, rootfs, monitor_memory=False)
+    vm.spawn()
+    vm.basic_config(vcpu_count=vcpus, mem_size_mib=1024)
```
Comment (kalyazin) [inline tests/integration_tests/performance/test_network_tcp_throughput.py:319]: Nit: and here.

Comment (pb8o) [reply tests/integration_tests/performance/test_network_tcp_throughput.py]: Should be gone now

File: tests/integration_tests/performance/test_network_latency.py (hunk 3 above)
Comment (pb8o) [reply tests/integration_tests/performance/test_network_latency.py]: Fixed now

**Review by kalyazin — APPROVED** [review]

**Review by kalyazin — APPROVED** [review]

**Review by andreitraistaru — APPROVED** [review]

**Pull request merged**